bump/bump_test.go

```diff
@@ -182,29 +187,72 @@ func TestBumpWithAnnotatedTags(t *testing.T) {
 			assert.Nil(t, err)
 
 			{
-				tag, err := Bump(r.Repo, "patch", Options{TagPrefix: prefix, VersionedBranches: mainBranch})
+				tag, err := Bump(r.Repo, "patch", Options{TagPrefix: prefix, VersionedBranches: mainBranch, FirstTagEncountered: false})
 				assert.Nil(t, err)
 				assert.Equal(t, NewVersionT(t, "0.0.1"), tag)
 			}
 
 			r.DoCommit("foo")
 			assert.Nil(t, CreateTag(r.Repo, "1.0.0", prefix, false))
 			{
-				tag, err := Bump(r.Repo, "patch", Options{TagPrefix: prefix, VersionedBranches: mainBranch})
+				tag, err := Bump(r.Repo, "patch", Options{TagPrefix: prefix, VersionedBranches: mainBranch, FirstTagEncountered: false})
 				assert.Nil(t, err)
 				assert.Equal(t, NewVersionT(t, "1.0.0"), tag)
 			}
 
 			r.DoCommit("bar")
 			{
-				tag, err := Bump(r.Repo, "patch", Options{TagPrefix: prefix, VersionedBranches: mainBranch})
+				tag, err := Bump(r.Repo, "patch", Options{TagPrefix: prefix, VersionedBranches: mainBranch, FirstTagEncountered: false})
 				assert.Nil(t, err)
 				assert.Equal(t, NewVersionT(t, "1.0.1"), tag)
 			}
 		})
 	}
 }
 
+//nolint:scopelint,paralleltest
+func TestBumpWithCheckoutNewBranchWithUntaggedCommitInHotFixBranch(t *testing.T) {
```

Comment:
A use case that might have to be tested. Even if manually.

If you branch using the nearest semantic version, and merge this in. Would this break the "latest" functionality. i,e 

Say main had the following strucutre 0.1.0 -> 0.2.0 -> 0.3.0

Off of 0.2.0 there is a branch with tag 0.2.1. If this were to merge back into master. Would the latest be 0.3.0 or the intended 0.4.0? Does this make sense?

Reply:
In that scenario, assuming that a merge to master bumps a minor version, that merge would create the tag 0.4.0.

Not understanding the use case and where it might fail 😕